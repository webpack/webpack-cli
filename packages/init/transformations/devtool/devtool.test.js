"use strict";

const defineTest = require("@webpack-cli/utils/defineTest");

defineTest(__dirname, "devtool", "devtool-0", "'source-map'", "init");
defineTest(__dirname, "devtool", "devtool-0", "myVariable", "init");
defineTest(
	__dirname,
	"devtool",
	"devtool-1",
	"'cheap-module-source-map'",
	"init"
);
defineTest(__dirname, "devtool", "devtool-1", "false", "init");

defineTest(__dirname, "devtool", "devtool-2", "'source-map'", "add");
defineTest(__dirname, "devtool", "devtool-3", "myVariable", "add");
defineTest(
	__dirname,
	"devtool",
	"devtool-3",
	"'cheap-module-source-map'",
	"add"
);
defineTest(__dirname, "devtool", "devtool-4", false, "add");
defineTest(__dirname, "devtool", "devtool-4", "false", "add");
