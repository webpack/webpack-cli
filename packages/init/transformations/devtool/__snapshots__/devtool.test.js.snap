// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`devtool transforms correctly using "devtool-0" data 1`] = `
"module.exports = {
    entry: 'index.js',

    output: {
		filename: 'bundle.js'
	},

    devtool: 'source-map'
}
"
`;

exports[`devtool transforms correctly using "devtool-0" data 2`] = `
"module.exports = {
    entry: 'index.js',

    output: {
		filename: 'bundle.js'
	},

    devtool: myVariable
}
"
`;

exports[`devtool transforms correctly using "devtool-1" data 1`] = `
"module.exports = {
    entry: 'index.js',

    output: {
		filename: 'bundle.js'
	},

    devtool: 'cheap-module-source-map'
}
"
`;

exports[`devtool transforms correctly using "devtool-1" data 2`] = `
"module.exports = {
    entry: 'index.js',

    output: {
		filename: 'bundle.js'
	},

    devtool: false
}
"
`;

exports[`devtool transforms correctly using "devtool-2" data 1`] = `
"module.exports = {
    entry: 'index.js',

    output: {
		filename: 'bundle.js'
	},

    devtool: 'source-map'
}
"
`;

exports[`devtool transforms correctly using "devtool-3" data 1`] = `
"module.exports = {
    entry: 'index.js',

    output: {
		filename: 'bundle.js'
	},

    devtool: myVariable
}
"
`;

exports[`devtool transforms correctly using "devtool-3" data 2`] = `
"module.exports = {
    entry: 'index.js',

    output: {
		filename: 'bundle.js'
	},

    devtool: 'cheap-module-source-map'
}
"
`;

exports[`devtool transforms correctly using "devtool-4" data 1`] = `
"module.exports = {
	entry: 'index.js',
	output: {
		filename: 'bundle.js'
	},
	devtool: false
}
"
`;

exports[`devtool transforms correctly using "devtool-4" data 2`] = `
"module.exports = {
	entry: 'index.js',
	output: {
		filename: 'bundle.js'
	},
	devtool: false
}
"
`;
